packages/cardano-services/src/StakePool/TypeormStakePoolProvider/TypeormStakePoolProvider.ts

```diff
@@ -50,9 +74,51 @@ export class TypeormStakePoolProvider extends TypeormProvider implements StakePo
     this.#lastRosEpochs = lastRosEpochs;
   }
 
+  async initializeImpl() {
+    await super.initializeImpl();
+
+    void this.withDataSource((dataSource) => this.getFuse(dataSource));
+  }
+
+  private async getFuse(dataSource: DataSource, fuzzyOptions?: FuzzyOptions) {
+    const {
+      threshold,
+      weights: { description, homepage, name, poolId, ticker }
+    } = this.#overrideFuzzyOptions ? { ...this.#fuzzyOptions, ...fuzzyOptions } : this.#fuzzyOptions;
+
+    const cacheKey = this.#overrideFuzzyOptions
+      ? `all-metadata-${JSON.stringify([description, homepage, name, threshold, ticker])}`
+      : 'all-metadata';
+
+    return this.#cache.get(cacheKey, async () => {
+      const metadata = await dataSource
+        .createQueryBuilder()
+        .from(PoolMetadataEntity, 'm')
```

Comment:
I think this will return multiple metadata per pool (if exists). Maybe we should query `StakePool->lastRegistration->metadata` instead?
```suggestion
        .from(StakePool, 'm')
```

Reply:
Nice catch! Thank you

Reply:
6deffdf484 should solve this